src/store/CategoricalDisplayStore.js

```diff
@@ -200,10 +208,10 @@ export default class CategoricalDisplayStore {
                 filterCategory(category);
             }
         }
-        if (searchHasResult) {
-            this.lastFilteredItems = filteredItems;
-            return filteredItems;
-        }
+
+        /** if any data found, then allow updating lastFilteredItems prop by found data */
+        this.lastFilteredItems = searchHasResult ? filteredItems : this.lastFilteredItems;
+
         return this.lastFilteredItems;
```

Comment:
Find a way to return `filteredItems` instead. As I mentioned before, returning `this.lastFilteredItems` will confuse people even though the code works